cvxpy/tests/test_KKT.py

```diff
@@ -5,175 +5,194 @@
+        sth.check_dual_domains(places)
+        sth.check_stationary_lagrangian(places)
         return sth
 
-    def test_sdp_1max(self):
+    def test_sdp_1max(self, places=4):
         sth = STH.sdp_1('max')
         sth.solve(solver='SCS')
-        sth.verify_objective(places=2)
-        sth.check_primal_feasibility(places=2)
-        sth.check_complementarity(places=2)
-        sth.check_dual_domains(places=2)
-        sth.check_stationary_lagrangian(places=2)
+        sth.check_primal_feasibility(places)
+        sth.check_complementarity(places)
+        sth.check_dual_domains(places)
+        sth.check_stationary_lagrangian(places)
         return sth
 
-    def test_sdp_2(self):
+    def test_sdp_2(self, places=3):
         # places is set to 3 rather than 4, because analytic solution isn't known.
         sth = STH.sdp_2()
         sth.solve(solver='SCS')
-        sth.verify_objective(places=2)
-        sth.check_primal_feasibility(places=2)
-        sth.verify_primal_values(places=2)
-        sth.check_complementarity(places=2)
-        sth.check_dual_domains(places=2)
-        sth.check_stationary_lagrangian(places=2)
+        sth.check_primal_feasibility(places)
+        sth.check_complementarity(places)
+        sth.check_dual_domains(places)
+        sth.check_stationary_lagrangian(places)
         return sth
 
 
-class StandardTestPCPs(BaseTest):
+class TestKKT_PCPs(BaseTest):
 
     def test_pcp_1(self, places: int = 2):
         sth = STH.pcp_1()
         sth.solve(solver='SCS')
-        sth.verify_objective(places)
         sth.check_primal_feasibility(places)
-        sth.verify_primal_values(places)
         sth.check_complementarity(places)
-        sth.verify_dual_values(places)
+        try:
+            sth.check_dual_domains(places)
+        except NotImplementedError:
+            pass
         sth.check_stationary_lagrangian(places)
         return sth
 
     def test_pcp_2(self, places: int = 2):
         sth = STH.pcp_2()
         sth.solve(solver='SCS')
-        sth.verify_objective(places)
         sth.check_primal_feasibility(places)
-        sth.verify_primal_values(places)
         sth.check_complementarity(places)
-        sth.verify_dual_values(places)
+        try:
+            sth.check_dual_domains(places)
+        except NotImplementedError:
+            pass
         sth.check_stationary_lagrangian(places)
         return sth
 
     def test_pcp_3(self, places: int = 2):
         sth = STH.pcp_3()
         sth.solve(solver='SCS')
-        sth.verify_objective(places)
         sth.check_primal_feasibility(places)
-        sth.verify_primal_values(places)
         sth.check_complementarity(places)
-        sth.verify_dual_values(places)
+        try:
+            sth.check_dual_domains(places)
+        except NotImplementedError:
+            pass
         sth.check_stationary_lagrangian(places)
         return sth
 
 
-class StandardTestFlags(BaseTest):
+class TestKKT_Flags(BaseTest):
```

Comment:
With these tests it's possible to use much more informative names. I suggest
* ``tf_1`` -> ``nsd_flag``
* ``tf_2`` -> ``psd_flag``
* ``tf_3`` -> ``nonneg_flag``
* ``tf_4`` -> ``nonpos_flag``

(It's fine for the other tests to have names like ``pcp_1``, ``pcp_2``, ``sdp_2``, etc.., because they're actually meant to test a lot of functionality at once.)

Separately, please make sure you have tests that include ``symmetric=True`` and ``diag=True``.

Reply:
Understood! Will change those naming conventions.

Also, isn't `symmetric=True` already checked for in the SDP tests? As far as `diag=True` is concerned, we do not support it currently because of how it's not possible to take derivatives w.r.t `diag` variables since they are represented internally as CSC matrices.

Reply:
There are two ways to work with PSD matrices in CVXPY. The first is ``X = cp.Variable(shape=(n,n), PSD=True)``. The second is to declare  ``X = cp.Variable(shape=(n,n), symmetric=True)`` and then explicitly constrain ``X >> 0``. The PSD tests probably only cover one of these. However, they are different for purposes of KKT checks.

It's also possible that a user wants a matrix to be symmetric, but they don't care if it's PSD or NSD. So there should be a test that features ``X = cp.Variable(shape=(n,n), symmetric=True)`` without any semidefinite constraints.

Reply:
Understood! I've added a new test (check `tf_3` in `test_KKT.py`) which has a variable with _just_ the `symmetric=True` flag

Reply:
Thanks. I still need you to address the unclear naming conventions in ``tf_1``, ``tf_2``, .... If someone runs ``pytest cvxpy/tests`` and some of these tests fail, then they should immediately know which parts of CVXPY's functionality is misbehaving (e.g., something related to PSD variables, or symmetric variables, or nonnegative variables).

Reply:
Done!